File: 04_linkedList.py
# Create a linked list in python


class node:
	def __init__(self, data=None):
		self.data=data
		self.next=None

class linkedList:
	def __init__(self):
		self.head=node()
	
	 

	def erase(self, index):
		if(index >= self.size()):
			print("ERROR: Index to large to append.")
			return None

		current_node = self.head
		current_index = 0

		while index != current_index:
			current_node = current_node.next
			current_index += 1

		current_node.next = current_node.next.next

	def get(self, index):
		if(index >= self.size()):
			print("ERROR: Index to large to append.")
			return4

		current_node = self.head
		current_index = 0

		while index != current_index:
			current_node = current_node.next
			current_index += 1

		return current_node.next.data
	
	def size(self):
		current_node = self.head
		length = 0

		while current_node.next != None:
			current_node = current_node.next
			length+=1
		return length

	def display(self):
		current_node = self.head
		index = 0

		while current_node.next != None:
			current_node = current_node.next
			print "Node", index, "is:", current_node.data
			index+=1

my_list = linkedList()
my_list.append(1)
my_list.append(2)
my_list.append(3)
my_list.append(4)
my_list.append(5)
my_list.display()
print "Size of list:", my_list.size()
my_list.erase(3)
my_list.display()

print "The 2 indice of the list is", my_list.get(3)

File: 04_LinkedList.c
#include <stdio.h>
#include <stdlib.h>

struct LinkedList{
	int data;
	struct LinkedList* next;
};

typedef struct LinkedList *node;


node createNode(){
	node temp;
	temp = (node)malloc(sizeof(struct LinkedList));
	temp->next = NULL;
	return temp;
}

node append(node head, int data){
	node temp, current_node;
	temp = createNode();
	temp->data = data;

	if(head == NULL){
		head=NULL;
	}else{
		current_node = head;

		while(current_node->next != NULL){
			current_node = current_node->next;
		}

		// We are at the last node in the linked list
		current_node->next = temp;
	}

	return head;
}

void displayList(node head){
	node current_node = head;

	while(current_node->next != NULL){
		current_node = current_node->next;
		printf("%d ", 	current_node->data);
	}
}

int length(node head){
	int index = 0;
	node current_node = head;

	while(current_node->next != NULL){
		current_node = current_node->next;
		index++;
	}

	return index;
}


int main(){

	node head = createNode();

	for(int i = 0; i <= 30; i+=5){
		append(head, i);
	}
	printf("Here is the newly created linked list: [ ");
	displayList(head);
	printf("]\n");

	printf("It is of size: %d\n", length(head));

	return 0;
}   